### PR TITLE
[Reviewer: Rob] Add in new get/delete options

### PR DESCRIPTION
**Pull request by eleanor-merry**
Rob, can you review these changes to support new uses of the send_get/send_delete methods

### CODE REVIEW

File: test_utils/fakecurl.cpp [hunk 1]
```diff
@@ -156,27 +156,27 @@ CURLcode FakeCurl::easy_perform(FakeCurl* curl)
   {
     // Call the debug callback with some dummy HTTP messages (to exercise SAS
     // logging code).
-    char* text;
+    std::string text;
```
Comment (rkd-msw) [inline test_utils/fakecurl.cpp]: Why have you made this change (just to cast it back to a char*)?

File: src/httpconnection.cpp [hunk 2]
```diff
@@ -417,22 +461,26 @@ HTTPCode HttpConnection::send_get(const std::string& path,                     /
...
                                   const std::string& username,                 //< Username to assert (if assertUser was true, else ignored)
+                                  std::vector<std::string> headers_to_add,     //< Extra headers to add to the request
                                   SAS::TrailId trail)                          //< SAS trail
 {
   CURL *curl = get_curl_handle();
 
   curl_easy_setopt(curl, CURLOPT_HTTPGET, 1);
 
-  return send_request(path, "", response, username, trail, "GET", curl);
+  std::vector<std::string> unused_req_headers;
```
Comment (rkd-msw) [inline src/httpconnection.cpp]: Isn't this unnecessary?

File: src/httpconnection.cpp [hunk 3]
```diff
@@ -288,6 +288,20 @@ HTTPCode HttpConnection::send_delete(const std::string& path,
 HTTPCode HttpConnection::send_delete(const std::string& path,
                                      SAS::TrailId trail,
                                      const std::string& body,
+                                     const std::string& override_server)
```
Comment (rkd-msw) [inline src/httpconnection.cpp]: What does override_server do?

File: test_utils/fakecurl.cpp (hunk 1 above)
Comment (eleanor-merry) [inline test_utils/fakecurl.cpp]: To stop Chronos whining

File: src/httpconnection.cpp (hunk 3 above)
Comment (rkd-msw) [inline src/httpconnection.cpp]: (Presumably it changes the underlying server used by this connection - should that be a separate method that makes its side effect very clear? `change_server`?)

File: test_utils/fakecurl.cpp [hunk 4]
```diff
@@ -156,27 +156,27 @@ CURLcode FakeCurl::easy_perform(FakeCurl* curl)
...
 
     text = "GET / HTTP/1.1\r\n\r\n";
     _debug_callback((CURL*)this,
                     CURLINFO_HEADER_OUT,
-                    text,
-                    strlen(text),
+                    (char*)text.c_str(),
+                    strlen(text.c_str()),
```
Comment (rkd-msw) [inline test_utils/fakecurl.cpp]: `text.size()`?

File: test_utils/fakecurl.cpp (hunk 1 above)
Comment (rkd-msw) [inline test_utils/fakecurl.cpp]: You could change `debug_callback_t` to take a `const char*`, or `strdup()` the const char* and free it at the end of the function.

**Comment by rkd-msw** [comment]
Looks fine. Some slightly perfectionist comments.

File: test_utils/fakecurl.cpp (hunk 4 above)
Comment (bossmc) [inline test_utils/fakecurl.cpp]: `text.length()`?

(jk - they're defined to be equal).

**Pull request merged**